compparallel_test.go

```diff
@@ -106,11 +104,53 @@ func TestComposableParallelFixtures(t *testing.T) {
 				{key: "a", value: "av", searchValCount: 2},
 			},
 			PutValue: []putValueFixture{
-				{err: errors.New("2 errors occurred:\n\t* a\n\t* a\n\n"), key: "/error/a", value: "a"},
+				{err: errors.New("a; a"), key: "/error/a", value: "a"},
 				{key: "a", value: "a"},
 			},
 			Provide: []provideFixture{{
-				err: errors.New("2 errors occurred:\n\t* routing: operation or key not supported\n\t* routing: operation or key not supported\n\n"),
+				err: errors.New("routing: operation or key not supported; routing: operation or key not supported"),
+			}},
+			FindPeer:    []findPeerFixture{{peerID: "pid1"}, {peerID: "pid3"}},
+			SearchValue: []searchValueFixture{{key: "a", vals: []string{"a", "a"}}},
+		},
+		{
+			Name: "simple two routers, one with delay, plus a third nothing DoNotWaitForStreamingResponses router",
+			routers: []*ParallelRouter{
+				{
+					Timeout:     time.Second,
+					IgnoreError: false,
+					Router: &Compose{
+						ValueStore:     newDummyValueStore(t, []string{"a", "b", "c"}, []string{"av", "bv", "cv"}),
+						PeerRouting:    newDummyPeerRouting(t, []peer.ID{"pid1", "pid2"}),
+						ContentRouting: Null{},
+					},
+				},
+				{
+					Timeout:      time.Minute,
+					IgnoreError:  false,
+					ExecuteAfter: time.Second,
+					Router: &Compose{
+						ValueStore:     newDummyValueStore(t, []string{"a", "d"}, []string{"av2", "dv"}),
+						PeerRouting:    newDummyPeerRouting(t, []peer.ID{"pid1", "pid3"}),
+						ContentRouting: Null{},
+					},
+				},
+				{
+					DoNotWaitForStreamingResponses: true,
+					IgnoreError:                    true,
+					Router:                         nothing{},
+				},
+			},
+			GetValue: []getValueFixture{
+				{key: "d", value: "dv", searchValCount: 1},
+				{key: "a", value: "av", searchValCount: 2},
+			},
+			PutValue: []putValueFixture{
+				{err: errors.New("a; a"), key: "/error/a", value: "a"},
+				{key: "a", value: "a"},
+			},
+			Provide: []provideFixture{{
+				err: errors.New("routing: operation or key not supported; routing: operation or key not supported"),
```

Comment:
One of the router is a `nothing`, you would expect to see `context.Canceled` but this does not happen since the api does not support streaming errors (so the channel is closed and that it).